acceptance/helpers.rb

```diff
@@ -16,16 +16,139 @@ def agents_only
       end
     end
 
+    module Structure
+      def collection_cmp(col_1, col_2)
+        col_1 = col_1.to_sym
+        col_2 = col_2.to_sym
+
+        return 0 if col_1 == col_2
+
+        return -1 if col_1 == 'pc1'
+        return 1 if col_2 == 'pc1'
+
+        col_1 <=> col_2
+      end
+      private :collection_cmp
+
+      # Restricts the test on the master's puppetserver collection. There are two
+      # types of restrictions that can be applied
+      #
+      # * Restrict the test to only run on a specific puppetserver collection. Here's
+      # an example of how you can specify this restriction:
+      #
+      # test_name 'an agent upgrade test' do
+      #   master_collection restrict_to: 'puppet5'
+      # end
+      #
+      # This will run the test if the master's puppetserver is from the puppet5 collection;
+      # otherwise, the test will be skipped.
+      #
+      # * Restrict the test to run on a valid range of collections. There are several ways
+      # you can specify this restriction:
+      #
+      # test_name 'an agent upgrade test' do
+      #   master_collection min: 'puppet4', max: 'puppet5'
+      # end
+      #
+      # test_name 'an agent upgrade test' do
+      #   master_collection min: 'puppet4'
+      # end
+      #
+      # test_name 'an agent upgrade test' do
+      #   master_collection max: 'puppet6'
+      # end
+      #
+      # The first restriction will only run the test if the master's puppetserver is from
+      # the puppet4 collection or newer, but no newer than puppet5. The second restriction
+      # will only run the test if the master's puppetserver is from the puppet4 collection
+      # or newer. The third restriction will only run the test if the master's puppetserver
+      # is from the puppet6 collection or older.
+      #
+      # NOTE: Be sure to put this DSL first in your test, before any confines. Otherwise,
+      # you could accidentally filter out your master and thus raise an exception here.
+      #
+      # @param [Hash] args The specified restriction
+      def master_collection(args)
```

Comment:
I think something akin to this DSL would be useful since we'll have a few tests that test collection-specific behavior (aside from the existing two that are already there). I just don't know if `master_collection` is the right name. The reason I based it off of the server version instead of the agent version as done previously is because the master's only involvement in agent upgrades is compiling the catalog, which is what the server does.

That being said, I think a name containing the word `collection` does make sense b/c it's how we think of upgrades.